### DISCUSSION TITLE
Raspberry Pi 4 Private Server >500ms Latency

### DISCUSSION THREAD

**Discussion by Helondeth** [jamulussoftware/jamulus#1396] [Technical Support & Help]
Hey together,
had a weird problem today.
I use a Raspberry Pi 4 (8GB with Rasperry Lite OS 64bit beta) as a Private Server.
Had a choir rehearsal last night and I set the numchannel to 30. 
Everything went well until the 11th Person logged in. Ping and Latency switched to >500ms instantly everywhere.
Today I changed the OS form „Pi OS Lite 64bit Beta“ to „Patchbox OS“ and I tried to access the server with more than 10 clients (I used several instances on my mac to check this)
It seems to work with 12. 
Do you think the OS is/was the Problem?

**Comment by dcorson-ticino-com** [comment]
I would use top to see the processor use.  It could be that the processor is maxing out.
I would, however, be a bit surprised as I run a server on an Odroid-N2 which is not that much more powerful than a Pi4 and have no problems with 25 users (Ubuntu 20.04).
Are you using the -T command line option to enable use of all the processor cores?
That may be the solution, but I think that 30 users on a Pi4 is being pretty optimistic.

**Reply by gene96817** [reply]
My original server was also a Raspberry Pi4 with Ubuntu 20.04. The best it could do was 14-15 singers with reasonable latency. Add one more person and it goes straight to >500ms. Looking at the system through nmon, the four cores were only 35-40% loaded. No hint of buffering problems or other resource issues. At the limit, 14 -15, it depends on the ping times of the participants. If one person has bad ping times, the limit could be 14. The difference between a 15 user limit and 25 user limit could be the ping times of the participants. I wish more is known about how Jamulus uses the hardware resources vs ping times.

**Reply by dcorson-ticino-com** [reply]
It could also be a router problem.  
With my server I see the ping times rise a bit as users are added, but it reaches a plateau after about 5 users.
What I see is the core usage mounting to about 40% and then the core frequency is increased a step.
This repeats as more users are added.
I have never seen the cores running at full speed, but I have never had more than 25 users on board and think that it will actually be my router which will be limiting as it runs much hotter than the Odroid-N2.

**Reply by gene96817** [reply]
I never suspected my router because it can handle speedtest at 1Gbps. That is far more than the expected traffic for 20 users and much more.

**Reply by dcorson-ticino-com** [reply]
I do not know, but I suspect that there is a big difference for the router doing bidirectional communications with 25 endpoints during a rehearsal and doing the unidirectional communication with one endpoint of a speed test. 
One of those being that it doesn't have much time to warm up during the one minute of a speed test compared to the more than 1 hour of a rehearsal.  I suspect that it has processor speed reduction because of overheating.
I have never noticed the router warming during a speed test, it tests to over 900Mb/s, after some time with 25 clients it is really hot.
But, as I said, I suppose it could be a problem if I try to push for more, up 'til now it is working OK, despite being very hot.

**Reply by gene96817** [reply]
very hot?   Your router is definitely being taxed. It will definitely lead to premature aging/failure or unreliable operations. The 25 endpoints are just some table lookups in the router operations. It's not much processing load, unless the memory chips are under-rated.  The router should be handling much less than 200 Mbps. It really shouldn't be a router problem unless you have a very inexpensive router.

**Reply by gilgongo** [reply]
Regarding routers: if you are regularly hosting larger groups on a domestic broadband connection, you can get into situations where network latency can be a problem regardless of available bandwidth. This is I believe to do with how your router manages  packets (and UDP fragmentation may also be related to this). In my case, I found that things became a lot of more stable when I [turned on SQM](https://www.bufferbloat.net/projects/bloat/wiki/More_about_Bufferbloat/#why-does-sqm-work-so-well).

**Comment by dcorson-ticino-com** [comment]
Maybe I should update the router.  It was not a cheapy, but was the first generation with fiber optic.

**Comment by gene96817** [comment]
Checking for a firmware update is a good idea. I would be interested in the manufacturing date of your router and the date of the latest firmware available. I am still concerned that your router runs hot. A fiber optic router shouldn't be having heat problems. Maybe you ISP will treat the heat as a reason for a warranty exchange.
